lib/core/context.c

```diff
@@ -172,12 +171,13 @@ void h2o_context_dispose(h2o_context_t *ctx)
 void h2o_context_request_shutdown(h2o_context_t *ctx)
 {
     ctx->shutdown_requested = 1;
-    if (ctx->globalconf->http1.callbacks.request_shutdown != NULL)
-        ctx->globalconf->http1.callbacks.request_shutdown(ctx);
-    if (ctx->globalconf->http2.callbacks.request_shutdown != NULL)
-        ctx->globalconf->http2.callbacks.request_shutdown(ctx);
-    if (ctx->globalconf->http3.callbacks.request_shutdown != NULL)
-        ctx->globalconf->http3.callbacks.request_shutdown(ctx);
+
+    h2o_linklist_t *conn_list[] = {&ctx->_active_conns, &ctx->_idle_conns};
+    H2O_CONN_LIST_FOREACH(h2o_conn_t * conn, conn_list, {
```

Comment:
I was initially not a fan of `H2O_CONN_LIST_FOREACH` (I understand that this came from the original WIP PR), but my resistance is weakening.

Rather, I wonder if this macro would become better if we could write like:
```c
H2O_CONN_LIST_FOREACH(h2o_conn_t * conn, {&ctx->active_conns, &ctx->_idle_conns}, {
    ...
});`
to avoid the cost of declaring a variable for naming the linklists.